code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -361,20 +371,17 @@ var/list/ai_list = list()
 		if(C)
 			C.post_status("shuttle")
 
-	return
-
-/mob/living/silicon/ai/proc/ai_cancel_call()
-	set category = "AI Commands"
-
-	if(isDead())
-		to_chat(src, "You can't send the shuttle back because you are dead!")
+/mob/living/silicon/ai/proc/ai_recall_shuttle()
+	if(!ismalf(src))
+		to_chat(usr, "You send the signal to recall the shuttle but nothing happens!")
 		return
-	if(istype(usr,/mob/living/silicon/ai))
-		var/mob/living/silicon/ai/AI = src
-		if(AI.control_disabled)
-			to_chat(src, "Wireless control is disabled!")
-			return
-	recall_shuttle(src)
+	var/datum/faction/malf/M = find_active_faction_by_member(mind.GetRole(MALF))
+	if(M?.stage != FACTION_ENDGAME)
+		to_chat(usr, "You need to initiate the takeover first")
```

Comment:
should probably use span class warn here, and add a period

Reply:
The use of `?.` is completely fine. You are mixing it up with the `:` operator